### DISCUSSION TITLE
how to modify values using integer-based indexing

### DISCUSSION THREAD

**Discussion by arfriedman** [pydata/xarray#7043] [Q&A] [answered]
Hi all,

I am enjoying using integer-based (isel) indexing to select DataArray coordinates as it's more explicit than numpy positional indexing. However, I'm confused about how to modify individual values of a DataArray using this type of indexing.

For example, suppose I have a DataArray da with coordinate dimensions lat and lon. 
I would like to do something like to set the value and the 1st latitude and 2nd longitude points to 10:
`da.isel(lat=0, lon=1) = 10`

However, according to the [documentation](https://docs.xarray.dev/en/stable/user-guide/indexing.html#assigning-values-with-indexing), this is not recommended:
> Do not try to assign values when using any of the indexing methods isel or sel:
> 
> # DO NOT do this
> `da.isel(space=0) = 0`
> 

I'm having difficulty finding the correct approach in the documentation. Is there a recomended method?

Thanks much,
Andrew

**Comment by mathause** [comment, marked as answer]
This can be done using `da[dict(lat=0, lon=1)] = 10`.

**Reply by arfriedman** [reply]
Thank you!

**Reply by arfriedman** [reply]
Perhaps it would be helpful to include this recommended approach in the documentation where it instructs not to assign values using `isel`? I do see that there is an example of integer-based dictionary assignment in the following subsection (`[87]`), but to me it was not initially clear that this addresses the warning in the red box above.

**Reply by dcherian** [reply]
@arfriedman A pull request to add that to the documeentation would be very welcome!

**Reply by arfriedman** [reply]
Will do!
